test/mocha/01-api.js

```diff
@@ -0,0 +1,270 @@
+/*!
+ * Copyright (c) 2022 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const {agent} = require('bedrock-https-agent');
+const fs = require('fs');
+const {httpClient} = require('@digitalbazaar/http-client');
+const path = require('path');
+
+const {readFile} = fs.promises;
+
+describe('httpClientHandler', () => {
+  const BASE_URL = `https://localhost:18443`;
+  it('configured route should work properly', async () => {
+    let res;
+    let err;
+    try {
+      res = await httpClient.get(`${BASE_URL}/test`, {agent});
+    } catch(e) {
+      err = e;
+    }
+    should.exist(res);
+    should.not.exist(err);
+    res.status.should.equal(200);
+    res.data.success.should.equal(true);
+  });
+  it('should handle error properly', async () => {
+    let res;
+    let err;
+    try {
+      res = await httpClient.get(`${BASE_URL}/unknown-error`, {agent});
+    } catch(e) {
+      err = e;
+    }
+    should.not.exist(res);
+    should.exist(err);
+    err.status.should.equal(400);
+    err.message.should.equal('An unknown error occurred.');
+  });
+  it('should handle "PermissionDenied" error properly', async () => {
+    let res;
+    let err;
+    try {
+      res = await httpClient.get(`${BASE_URL}/permission-denied-error`,
+        {agent});
+    } catch(e) {
+      err = e;
+    }
+    should.not.exist(res);
+    should.exist(err);
+    err.status.should.equal(403);
+    err.message.should.equal('Permission denied.');
+  });
+  it('should handle "NotFound" error properly', async () => {
+    let res;
+    let err;
+    try {
+      res = await httpClient.get(`${BASE_URL}/not-found-error`, {agent});
+    } catch(e) {
+      err = e;
+    }
+    should.not.exist(res);
+    should.exist(err);
+    err.status.should.equal(404);
+    err.message.should.equal('Not Found.');
+    err.response.headers.get('content-type')
+      .should.equal('application/json; charset=utf-8');
+  });
+  it('should handle non-bedrock error properly', async () => {
+    let res;
+    let err;
+    try {
+      res = await httpClient.get(`${BASE_URL}/non-bedrock-error`, {agent});
+    } catch(e) {
+      err = e;
+    }
+    should.not.exist(res);
+    should.exist(err);
+    err.status.should.equal(500);
+  });
+  it('should handle "text/html" accept header', async () => {
+    let res;
+    let err;
+    try {
+      res = await httpClient.get(`${BASE_URL}/unhandled-error`, {
+        agent,
+        headers: {
+          Accept: 'text/html'
+        }
+      });
+    } catch(e) {
+      err = e;
+    }
+    should.not.exist(res);
+    should.exist(err);
+    err.status.should.equal(404);
+    err.message.should.equal('Not Found');
+    err.response.headers.get('content-type')
+      .should.equal('text/plain');
+  });
+  it('should serve a static file property with file and cors', async () => {
```

Comment:
These test names that say "it should serve..." are good and match up with my suggestions above w/context name change, etc. If you can make the others match with these that would be great.